### PR TITLE
Sample deduplication bot that demonstrates programmatic linking and w…

### PR DESCRIPTION
**Pull request by reshmakh**
…arning scenarios

### CODE REVIEW

File: src/examples/patient-deduplication.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+    if (
+      firstName === existingPatient.name?.[0].given?.[0].toString() &&
+      lastName === existingPatient.name?.[0].family?.toString() &&
+      birthDate === existingPatient.birthDate?.toString()
+    ) {
+      existingPatient.link = [
+        {
+          type: 'replaces',
+          other: {
```
Comment (codyebberson) [inline src/examples/patient-deduplication.ts:28]: You can use: `other: createReference(patient)`, which does exactly the same thing.

File: src/examples/patient-deduplication.test.ts [hunk 2]
```diff
@@ -0,0 +1,170 @@
+import { Patient } from '@medplum/fhirtypes';
+import { MockClient } from '@medplum/mock';
+import { handler } from './patient-deduplication';
+
+describe('Link Patient', async () => {
+  test('Success', async () => {
+    const medplum = new MockClient();
+    //Create an original Patient
```
Comment (codyebberson) [inline src/examples/patient-deduplication.test.ts:8]: Total nit, we should setup eslint and prettier on this repo.  There should be a space after `//`.

File: src/examples/patient-deduplication.test.ts [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+          given: ['Jane'],
+        },
+      ],
+    });
+
+    const contentType = 'application/fhir+json';
+    await handler(medplum, { input: patient2, contentType, secrets: {} });
+
+    console.log(JSON.stringify(patient2));
+    //expect(patient2.active).toBe(true);
```
Comment (codyebberson) [inline src/examples/patient-deduplication.test.ts:167]: In this case, you will have to re-read the `Patient` resource before you can expect changes to take effect.

File: src/examples/patient-deduplication.ts [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+export async function handler(medplum: MedplumClient, event: BotEvent): Promise<any> {
+  //This bot should only be triggered by a Patient resource Subscription only
+  const patient = event.input as Patient;
+  if (patient.resourceType !== 'Patient') {
+    throw new Error('Unexpected input. Expected Patient.');
+  }
+
+  const identifier = patient.identifier?.[0].value?.toString();
+
+  const existingPatient = await medplum.searchOne('Patient', 'identifier=' + identifier);
```
Comment (codyebberson) [inline src/examples/patient-deduplication.ts:13]: You probably want a filter to make sure you're searching for patients `!=` the input.

I.e. (untested)

```ts
medplum.searchOne('Patient', 'identifier=' + identifier + '&_id:not=' + patient.id);
```

File: src/examples/patient-deduplication.ts [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+          },
+        },
+      ];
+
+      //Save the linkage to the existing patient
+      medplum.updateResource(existingPatient);
+
+      //Mark the new patient as inactive
+      patient.active = false;
+      medplum.updateResource(patient);
```
Comment (codyebberson) [inline src/examples/patient-deduplication.ts:40]: We should also add ESLint "no floating promises" -- this call needs `await`, and could cause unexpected behavior.

File: src/examples/patient-deduplication.ts [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+    } else {
+      // This case is only an identifier collision, so this should be handled manually
+      console.log('Warning: Potential duplicate identifiers found, alert operator.');
+      // Mark the new patient as active to ensure it is not deleted or merged accidentally
+      patient.active = true;
+      medplum.updateResource(patient);
```
Comment (codyebberson) [inline src/examples/patient-deduplication.ts:46]: We should also add ESLint "no floating promises" -- this call needs `await`, and could cause unexpected behavior.

File: src/examples/patient-deduplication.test.ts (hunk 2 above)
Comment (reshmakh) [reply src/examples/patient-deduplication.test.ts:8]: Not really understanding why this doesn't work https://github.com/medplum/medplum-demo-bots/blob/main/package.json#L11

File: src/examples/patient-deduplication.test.ts [hunk 7]
```diff
@@ -0,0 +1,172 @@
...
+      name: [
+        {
+          family: 'Smith',
+          given: ['John'],
+        },
+      ],
+    });
+
+    const existingPatient = await medplum.search('Patient', 'identifier=999-47-5984');
+    console.log('Existing' + JSON.stringify(existingPatient, null, 2));
```
Comment (reshmakh) [inline src/examples/patient-deduplication.test.ts:53]: This produces empty search results

File: src/examples/patient-deduplication.ts (hunk 4 above)
Comment (rahul1) [reply src/examples/patient-deduplication.ts:13]: Bump this ^ - right now you're relying on `searchOne` not to return the input patient, which may or may not be true

**Review by rahul1 — APPROVED** [review]

**Comment by reshmakh** [comment]
This is related to https://github.com/medplum/medplum/issues/1460

**Pull request merged**